discordbot/OverworldDiscordBot.py

```diff
@@ -0,0 +1,282 @@
+                        if len(_event_channel_id) > 0:
+                            _channels_to_notify.append(_event_channel_id)
+                        _text_channel_id = self.config["TextNotificationChannelIdForMark"]
+                        if len(_text_channel_id) > 0:
+                            _message = "{} Mark {} detected!".format(get_pkm_mark_string(pkm), get_pkm_species_string(pkm))
+                            asyncio.run_coroutine_threadsafe(self.send_discord_msg(_message, Channels.TextNotificationChannelIdForMark),self.loop)
+                    
+                    # Remove duplicate channel IDs from the list.
+                    _channels_to_notify = list(set(_channels_to_notify))
+                    if _channels_to_notify and strtobool(self.config["EnableDebugLogging"]):
+                        print("DEBUG: Channels to notify are: {}".format(_channels_to_notify))
+
+                    # If we are notifying at least one channel ...
+                    if _channels_to_notify:
+                        # if so, format to strings
+                        title, description = pkm_format(pkm, ctx)
+                        # create discord embed object with the color specified
+                        embed=discord.Embed(color=embed_color)
+                        # add pkm strings as a field
+                        embed.add_field(name = title, value = description, inline=False)
+                        with io.BytesIO() as image_binary:
+                            # pull pokemon image
+                            get_pokemon(pkm.species, pkm.shinyType).save(image_binary, 'PNG')
+                            image_binary.seek(0)
+                            # save to a discord file
+                            file = discord.File(fp=image_binary, filename='image.png')
+                        # set embed thumbnail to the attached discord file
+                        embed.set_thumbnail(url="attachment://image.png")
+                        
+                        # send a message in channel start was called with the infomation of the pokemon
+                        # asyncio.run_coroutine_threadsafe(ctx.send(embed=embed,file=file),self.loop)
+                        for channel_id in _channels_to_notify:
+                            asyncio.run_coroutine_threadsafe(self.send_discord_event(embed, file, channel_id),self.loop)
+                # print a line to show that new pokemon have just been read
+                _message = "{} Pokemon Observed".format(len(pkms))
+                asyncio.run_coroutine_threadsafe(self.send_discord_msg(_message, Channels.NotificationChannelForInfo),self.loop)
+            # give the thread a break
+            self.reader.pause(0.3)
+
+def is_pkm_shiny(pkm):
+    return pkm.shinyType != 0
+
+def is_pkm_brilliant(pkm):
+    return pkm.brilliant
+
+def is_pkm_marked(pkm):
+    return pkm.mark > 52 and pkm.mark < 98 and pkm.mark != 69
```

Comment:
\> 52 and < 98 checks aren't necessarily needed as an overworld PK8 will not spawn with a ribbon, ``pkm.mark != 255`` is simpler